urllib3/response.py

```diff
@@ -475,8 +485,18 @@ def read_chunked(self, amt=None, decode_content=None):
                 if self.chunk_left == 0:
                     break
                 chunk = self._handle_chunk(amt)
-                yield self._decode(chunk, decode_content=decode_content,
-                                   flush_decoder=True)
+                decoded = self._decode(chunk, decode_content=decode_content,
+                                       flush_decoder=False)
+                if decoded:
+                    yield decoded
+
+            if decode_content:
+                # On CPython and PyPy, we should never need to flush the
+                # decoder. However, on Jython we *might* need to, so
+                # lets defensively do it anyway.
+                decoded = self._flush_decoder()
+                if decoded:  # Platform-specific: Jython.
```

Comment:
Because the flush works on all platforms, but it doesn't return anything on any other platform (the decoder was flushed at the last decompress call). The `if` block is not platform specific because `decode_content` may be `True` on any platform.